File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int maxPro = 0;
        int n = prices.size();
        int l = 0, r = 1;
        while( r < n){
            if( prices[r] > prices[l] ){
                int p = prices[r] - prices[l];
                maxPro = max( p, maxPro);
            }
            else{
                l = r;
            }
            r++;
        }
        return maxPro;
    }
};

File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int n = prices.length;
        int l=0, r=1;
        int maxPro = 0;
        while( r< n){
            if( prices[l] > prices[r]){
                l = r;
            }
            else{
                maxPro = Math.max( maxPro, prices[r]-prices[l]);
            }
            r++;
        }
        return maxPro;
    }
}